src/table/utils/__tests__/styling-utils.spec.ts

```diff
@@ -339,64 +342,64 @@ describe('styling-utils', () => {
   });
 
   describe('getColumnStyle', () => {
-    let styling;
-    let qAttrExps;
-    let stylingInfo;
+    let styling: CellStyle;
+    let qAttrExps: EngineAPI.INxAttributeExpressionValues;
+    let stylingIDs: string[];
 
     beforeEach(() => {
-      styling = { color: 'someFontColor' };
+      styling = { color: 'someFontColor', backgroundColor: 'someBgColor' };
       qAttrExps = {
-        qValues: [{ qText: '#dddddd' }, { qText: '#111111' }],
+        qValues: [
+          { qText: '#dddddd', qNum: NaN },
+          { qText: '#111111', qNum: NaN },
+        ],
       };
-      stylingInfo = ['cellBackgroundColor', 'cellForegroundColor'];
+      stylingIDs = ['cellBackgroundColor', 'cellForegroundColor'];
     });
 
     it('should return styling with both new fontColor and backgroundColor when selected', () => {
-      const columnStyle = getColumnStyle(styling, qAttrExps, stylingInfo);
+      const columnStyle = getColumnStyle(styling, qAttrExps, stylingIDs);
       expect(columnStyle.backgroundColor).toBe('rgb(221,221,221)');
       expect(columnStyle.color).toBe('rgb(17,17,17)');
     });
     it('should return styling with new fontColor', () => {
       qAttrExps.qValues = [qAttrExps.qValues[1]];
-      stylingInfo = [stylingInfo[1]];
+      stylingIDs = [stylingIDs[1]];
 
-      const columnStyle = getColumnStyle(styling, qAttrExps, stylingInfo);
+      const columnStyle = getColumnStyle(styling, qAttrExps, stylingIDs);
       expect(columnStyle.backgroundColor).toBe(undefined);
       expect(columnStyle.color).toBe('rgb(17,17,17)');
     });
     it('should return styling with backgroundColor', () => {
       qAttrExps.qValues = [qAttrExps.qValues[0]];
-      stylingInfo = [stylingInfo[0]];
+      stylingIDs = [stylingIDs[0]];
 
-      const columnStyle = getColumnStyle(styling, qAttrExps, stylingInfo);
+      const columnStyle = getColumnStyle(styling, qAttrExps, stylingIDs);
       expect(columnStyle.backgroundColor).toBe('rgb(221,221,221)');
       expect(columnStyle.color).toBe(STYLING_DEFAULTS.FONT_COLOR);
     });
     it('should return styling unchanged', () => {
-      qAttrExps = undefined;
```

Comment:
i think it is also good to check qAttrExps is something then should return styling unchanged

Reply:
what do you mean? if qAttrExps is set then you actually have column styling and it should change, that is what the tests above do

Reply:
actually we should probably have another test that tests what happens when you qAttrExps is undefined

Reply:
yes, I am considering adding a new test case to cover a case when qAttrExps is not set or set something else then it should return unchanged styling

Reply:
so I tested what the layout becomes if you only add the id but no expression in the properties. You get a layout with qAttrExps and it has qValues as well. but it does not have qText. So need to change the logic a bit so it does not change anything unless there is a qText

Reply:
Turns out that getColumnStyle never runs if stylingIDs is empty. So no need to sanity check for that. We do need to check that qText is there and it resolved to a valid color. So added checks + tests for that instead

Reply:
Nice